Add recipe for imgur

KeyWeeUsr opened the pull request:
### Brief summary of what the package does

This package provides an unofficial Emacs client for Imgur API allowing authorization, upload and deletion with multiple sessions with both non-interactive functions and interactive commands.

### Direct link to the package repository

https://github.com/KeyWeeUsr/imgur

### Your association with the package

Maintainer

### Relevant communications with the upstream package maintainer

**None needed**

### Checklist

<!-- Please confirm by replacing `[]` with `[x]`: -->

- [X] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [X] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [X] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [X] My elisp byte-compiles cleanly
- [X] I've used `M-x checkdoc` to check the package's documentation strings
- [X] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)

<!-- After submitting, please fix any problems the CI reports. -->

riscy commented:
Just so you're aware, there is an imgur in the attic - https://github.com/emacsattic/imgur/blob/master/imgur.el - but I think it's outdated by this point.  Some quick lints -

- imgur.el#L283: Avoid `with-no-warnings` if the root cause can be addressed
- imgur.el#L410: `read-string` prompts should not be used for this purpose; maybe a `y-or-n` prompt makes more sense here
- Reminder: ensure GitHub release 1.0.0 is up-to-date with your current code and `Package-Version` after we make changes here

KeyWeeUsr commented:
@riscy Thanks for the attic link, it certainly has an interesting approach. I could add a minor mode similar to it to check for dired or other modes :thinking:
The `with-no-warnings` was only due to an old unibyte func but something, probably json, required 27.1+ so I dropped the old func. Hopefully I fixed the rest. The tag was bumped to the latest commit :+1:

riscy commented:
I think acknowledging the earlier package in your README is a good idea and - as you are already doing - considering any features you could port over is also a good idea.  But I think merging this one (which has no curl dependency for instance) to MELPA is the right move.

As a reminder, please ensure your GitHub release is up-to-date with your current code and `Package-Version` after we merge here.
